tests/network/primary_pod_network.go

```diff
@@ -107,7 +107,10 @@ var _ = SIGDescribe("[Serial]Primary Pod Network", func() {
 				It("should report PodIP/s as its own on interface status", func() {
 					vmiPod := tests.GetRunningPodByVirtualMachineInstance(vmi, vmi.Namespace)
 					Eventually(vmiIP).Should(WithTransform(removeCIDR, Equal(vmiPod.Status.PodIP)), "should contain VMI Status IP as Pod status ip")
-					Eventually(vmiIPs).Should(ContainElements(vmi.Status.Interfaces[0].IP, libnet.DefaultIPv6Address), "should contain at IPs the IP and the cloud-init configured ipv6")
+
+					// Check first fixed IPv6 address so we are sure that GA has quick in
+					Eventually(vmiIPs).Should(ContainElement(libnet.DefaultIPv6Address), "should contain IPv6 address set by cloud-init and reported by guest agent")
```

Comment:
I know this is not your doing, but help me understand the test.

The purpose of this is to check `It("should report PodIP/s as its own on interface status"`. Why do we check here that we report the "fake" IP address? Does not seem to be the purpose of this tests and it is misleading. But maybe I'm getting this wrong.

Reply:
In other words - this test checks that we report PodIPs on the VM. You verify the opposite. Do we have a bug backed up by an invalid test or am I reading this wrong?

Reply:
Well maybe the `It` message has to be augmented, we added this to check that cloud-init network data generated from test is really working fine.

Reply:
To me it sounds that you should create a different test to cover the new functionality.

The name of the test is part of our test-plan and outlining how KubeVirt handles IP. Maybe I miss the logic why checking of static IP in cloud init is related to our mechanism of copying of the Pod IP to the VM?

Reply:
Having Pod IP in the VM Status is part of the story I suppose we also has what is expected at VMI IPs field right ? and for that case we are having the IPv6 from VMI and the IPv4 from pod.

Reply:
> IPv6 from VMI and the IPv4 from pod

Eventually we should get IPv6 from pod as well. This temporary gap will be resolved once we have a DHCP server implemented for IPv6.

Could you please specify the title of this test to talk about IPv4 and introduce a new one that describes what currently happens with IPv6 (that it is statically configured). It would illustrate the current way nicer. Once we have the DHCP ready, we would adjust the plan.

Would it make sense?

Reply:
Totally, thanks @phoracek